### PR TITLE
feat: ensure filter label is defined when classifying filters

### PR DESCRIPTION
**Pull request by Averethel**
References: CAR-7635

When defining filters it is now required to define how a label for the filter is generated.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -357,19 +375,31 @@ To this end `deriveFilters` method provided by the package can be used. It is a
...
 const definition = {
-  isManual: ({ filters }) => {
-    switch (filters?.listingType) {
-      case ListingType.Manual:
-        return true
-      case ListingType.Imported:
-        return false
-    }
+  isManual: {
+    derive: ({ filters }) => {
```
Comment (masone) [inline README.md:389]:
```suggestion
    getValue: ({ filters }) => {
```

for the sake of symmetry? `derive` is a bit abstract to me

File: README.md [hunk 2]
```diff
@@ -235,7 +235,11 @@ This package provides a way to group related search parameters (classify them).
...
 
-This package provides `classifyParams` that classifies the parameters and a dual `toDecodedParam` function that reverses (flattens) the classification. `classifyParams` is a higher-order function that requires classification to be passed to it. This is useful if you pla to reuse the classification function within the project. Values considered as empty (`null`, `undefined`, `""` and `[]`) will be removed during classification.
+Since it's desirable to know how to render filter value as a tag either to visualize applied filters better or to enable clearing single filters more easily when you want to classify a parameter as a filter, you need to provide `getLabel` method as well. It takes the current filter value as an argument and returns a string or an array of strings (think about multiple selection filters). You can also pass an optional argument containing:
+- `t` - translation function
+- `mappings` - a collection of function maps keys to specific values (think `makeKey` - `make.name` mapping)
```
Comment (masone) [inline README.md:240]: can you add an example for the usage of mappings as well please?

File: README.md [hunk 3]
```diff
@@ -263,10 +267,24 @@ const classification = {
...
+    getLabel: (value, { t }) =>
+      value ? t("hasImages") : t("noImages"),
+  },
+  listingType: {
+    kind: "filters",
+    getLabel: (value, { t }) =>  t(`listingTypes.${value}`),
+  },
+  powerTo: {
+    kind: "filters",
+    getLabel: ({ value, unit }) => `max: ${value} ${unit}`,
```
Comment (masone) [inline README.md:286]: is the signature of this example off? value would be the first positional attribute, no? where is unit coming from?
```suggestion
    getLabel: (value) => `max: ${value} PS`,
```

File: src/paramsClassification/index.ts [hunk 4]
```diff
@@ -33,7 +33,11 @@ export const classifyParams = <T>(
   const classifiedQuery = buildEmptyQuery<T>()
 
   Object.entries(params).forEach(([name, value]) => {
-    const paramClassification = paramsClassification[name] || otherGroup
+    const classificationKind = paramsClassification[name] || otherGroup
+    const paramClassification =
+      typeof classificationKind === "string"
+        ? classificationKind
+        : classificationKind.kind
```
Comment (masone) [inline src/paramsClassification/index.ts:40]: according to the types, it would always have a `kind`. what's the case where it would be a string?

**Review by masone — COMMENTED** [review]
megusta! relatively simple change that unlocks a lot of potential

File: README.md (hunk 3 above)
Comment (Averethel) [reply README.md:286]: No, it's encoded value in the definition above: `150|PS` => `{ value: 150, unit: "PS" }`
`getLabel` always takes decoded values as argument.

File: src/paramsClassification/index.ts (hunk 4 above)
Comment (Averethel) [reply src/paramsClassification/index.ts:40]: No, we define with plain strings when it's simple so:
```
page: "pagination"
```
but:
```
makeModelType: { 
  kind: "filter"
  getLabel: ....
}
```

I thought about always using object with `kind` but it looked way too verbose.

File: README.md (hunk 3 above)
Comment (masone) [reply README.md:286]: ooh I see, the value is an object. that filter returns a special object, not just a string value...

Comment (masone) [reply README.md:286]: (I find the example confusing without remembering these details, maybe worth an inline comment?)

**Comment by Averethel** [comment]
@masone care to take another look?

**Review by masone — APPROVED** [review]
swell! thanks for the explanations

**Pull request merged**